Exception when calling WebAuthClient.cancel()

### Describe the bug?

When `WebAuthClient.cancel()` is called on some Android versions, the `onError` callback is invoked.
I've found this when testing on emulators. I don't have the devices on hand to test whether it happens on real devices as well.

### What is expected to happen?

The `onCancel` callback is invoked.

### What is the actual behavior?

The following exception:

Android 8, 8.1, 9, 10, 12
```
AuthorizationException: {"type":0,"code":0,"errorDescription":"Socket closed"}
        at com.okta.oidc.net.request.ConfigurationRequest.executeRequest(ConfigurationRequest.java:68)
        at com.okta.oidc.clients.AuthAPI.obtainNewConfiguration(AuthAPI.java:101)
        at com.okta.oidc.clients.web.SyncWebAuthClientImpl.signIn(SyncWebAuthClientImpl.java:222)
        at com.okta.oidc.clients.web.WebAuthClientImpl.lambda$signIn$3$com-okta-oidc-clients-web-WebAuthClientImpl(WebAuthClientImpl.java:149)
        at com.okta.oidc.clients.web.WebAuthClientImpl$$ExternalSyntheticLambda1.run(Unknown Source:6)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.net.SocketException: Socket closed
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:394)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:230)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:212)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
        at java.net.Socket.connect(Socket.java:621)
        at com.android.okhttp.internal.Platform.connectSocket(Platform.java:182)
        at com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:145)
        at com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:116)
        at com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:186)
        at com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:128)
        at com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:97)
        at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:289)
        at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:232)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:465)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:131)
        at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.connect(DelegatingHttpsURLConnection.java:90)
        at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:30)
        at com.okta.oidc.net.HttpClientImpl.connect(HttpClientImpl.java:113)
        at com.okta.oidc.net.request.BaseRequest.openConnection(BaseRequest.java:58)
        at com.okta.oidc.net.request.ConfigurationRequest.executeRequest(ConfigurationRequest.java:60)
        at com.okta.oidc.clients.AuthAPI.obtainNewConfiguration(AuthAPI.java:101) 
        at com.okta.oidc.clients.web.SyncWebAuthClientImpl.signIn(SyncWebAuthClientImpl.java:222) 
        at com.okta.oidc.clients.web.WebAuthClientImpl.lambda$signIn$3$com-okta-oidc-clients-web-WebAuthClientImpl(WebAuthClientImpl.java:149) 
        at com.okta.oidc.clients.web.WebAuthClientImpl$$ExternalSyntheticLambda1.run(Unknown Source:6) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
```

Or sometimes
```
AuthorizationException: {"type":0,"code":0,"errorDescription":"Canceled"}
        at com.okta.oidc.net.request.ConfigurationRequest.executeRequest(ConfigurationRequest.java:68)
        at com.okta.oidc.clients.AuthAPI.obtainNewConfiguration(AuthAPI.java:101)
        at com.okta.oidc.clients.web.SyncWebAuthClientImpl.signIn(SyncWebAuthClientImpl.java:228)
        at com.okta.oidc.clients.web.WebAuthClientImpl.lambda$signIn$3$com-okta-oidc-clients-web-WebAuthClientImpl(WebAuthClientImpl.java:149)
        at com.okta.oidc.clients.web.WebAuthClientImpl$$ExternalSyntheticLambda1.run(Unknown Source:6)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.io.IOException: Canceled
        at com.okta.oidc.net.request.BaseRequest.openConnection(BaseRequest.java:60)
        at com.okta.oidc.net.request.ConfigurationRequest.executeRequest(ConfigurationRequest.java:60)
```

### Reproduction Steps?

- Construct a WebAuthClient client and call `signIn`
- After some delay, call `cancel`
- Exception

```
val client: WebAuthClient = buildWebClient(activity)
client.registerCallback(...)
client.signIn(activity, null)

GlobalScope.launch {
                // Delay simulates human reaction time. Immediately calling cancel causes other errors.
               // May need to adjust this to make sure the call happens before the custom tab is launched.
                delay(300L)
                cancel()
            }
```

### Additional Information?

_No response_

### SDK Version

1.2.5

### Build Information

Android 8, 8.1, 9, 10, 12

--- a/library/src/main/java/com/okta/oidc/clients/web/SyncWebAuthClientImpl.java
+++ b/library/src/main/java/com/okta/oidc/clients/web/SyncWebAuthClientImpl.java
@@ -238,6 +238,9 @@ class SyncWebAuthClientImpl extends AuthAPI implements SyncWebAuthClient {
             StateResult authResult = startSignIn(activity, request);
             return processSignInResult(authResult);
         } catch (AuthorizationException e) {
+            if (Thread.currentThread().isInterrupted()) {
+                return Result.cancel();
+            }
             return Result.error(e);
         } catch (IOException | InterruptedException e) {
             return Result.cancel();

